var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -148,32 +154,32 @@ class Llvm(CMakePackage):
     # https://bugs.llvm.org/show_bug.cgi?id=39696
     patch('thread-p9.patch', when='@develop+libcxx')
 
-    @run_before('cmake')
-    def check_darwin_lldb_codesign_requirement(self):
-        if not self.spec.satisfies('+lldb platform=darwin'):
-            return
-        codesign = which('codesign')
-        mkdir('tmp')
-        llvm_check_file = join_path('tmp', 'llvm_check')
-        copy('/usr/bin/false', llvm_check_file)
-
-        try:
-            codesign('-f', '-s', 'lldb_codesign', '--dryrun',
-                     llvm_check_file)
-
-        except ProcessError:
-            # Newer LLVM versions have a simple script that sets up
-            # automatically
-            setup = Executable("./lldb/scripts/macos-setup-codesign.sh")
+    if '+code_signing' in self.spec:
```

Comment:
`self` isn't defined at this point in the code, you should move this if-statement inside of `codesign_check`.

Reply:
This was what I thought caused the issue. Thanks for checking this @adamjstewart !

Reply:
Is spec defined at that point ? I also rebased everything on to develop.

Reply:
Nothing is defined at that point. You need to move the if-statement inside of the method definition to have access to `self` and `self.spec`.

Reply:
Done.